lib/web_ui/lib/src/engine/history/history.dart

```diff
@@ -22,80 +22,59 @@ part of engine;
 ///  * [MultiEntriesBrowserHistory]: which creates a set of states that records
 ///    the navigating events happened in the framework.
 abstract class BrowserHistory {
+  static BrowserHistory defaultImpl({required UrlStrategy? urlStrategy}) {
+    return MultiEntriesBrowserHistory(urlStrategy: urlStrategy);
+  }
+
   late ui.VoidCallback _unsubscribe;
 
   /// The strategy to interact with html browser history.
-  LocationStrategy? get locationStrategy => _locationStrategy;
-  LocationStrategy? _locationStrategy;
-  /// Updates the strategy.
-  ///
-  /// This method will also remove any previous modifications to the html
-  /// browser history and start anew.
-  Future<void> setLocationStrategy(LocationStrategy? strategy) async {
-    if (strategy != _locationStrategy) {
-      await _tearoffStrategy(_locationStrategy);
-      _locationStrategy = strategy;
-      await _setupStrategy(_locationStrategy);
-    }
-  }
+  UrlStrategy? get urlStrategy;
 
-  Future<void> _setupStrategy(LocationStrategy? strategy) async {
-    if (strategy == null) {
-      return;
-    }
-    _unsubscribe = strategy.onPopState(onPopState as dynamic Function(html.Event));
-    await setup();
-  }
+  bool _isDisposed = false;
 
-  Future<void> _tearoffStrategy(LocationStrategy? strategy) async {
-    if (strategy == null) {
-      return;
-    }
-    _unsubscribe();
-
-    await tearDown();
+  void _setupStrategy(UrlStrategy strategy) {
+    _unsubscribe = strategy.addPopStateListener(
+      onPopState as html.EventListener,
+    );
   }
 
   /// Exit this application and return to the previous page.
   Future<void> exit() async {
-    if (_locationStrategy != null) {
-      await _tearoffStrategy(_locationStrategy);
+    if (urlStrategy != null) {
+      await tearDown();
       // Now the history should be in the original state, back one more time to
       // exit the application.
-      await _locationStrategy!.back();
-      _locationStrategy = null;
+      await urlStrategy!.go(-1);
     }
   }
 
   /// This method does the same thing as the browser back button.
   Future<void> back() {
-    if (locationStrategy != null) {
-      return locationStrategy!.back();
+    if (urlStrategy != null) {
+      return urlStrategy!.go(-1);
```

Comment:
I think this can be simplified to:

```dart
FutureOr<void> back() {
  return urlStrategy?.go(-1);
}
```

Or


```dart
Future<void> back() async {
  return urlStrategy?.go(-1);
}
```